js_modules/dagster-ui/packages/ui-core/src/overview/useGroupTimelineRunsBy.tsx

```diff
@@ -1,45 +1,51 @@
-import {useCallback, useMemo} from 'react';
+import {useCallback} from 'react';
 
-import {useQueryAndLocalStoragePersistedState} from '../hooks/useQueryAndLocalStoragePersistedState';
+import {useQueryPersistedState} from '../hooks/useQueryPersistedState';
+import {useStateWithStorage} from '../hooks/useStateWithStorage';
 
 const GROUP_BY_KEY = 'dagster.run-timeline-group-by';
 
 export type GroupRunsBy = 'job' | 'automation';
 
+const validate = (value: string) => {
+  switch (value) {
+    case 'job':
+    case 'automation':
+      return value;
+    default:
+      return null;
+  }
+};
+
 export const useGroupTimelineRunsBy = (
   defaultValue: GroupRunsBy = 'job',
 ): [GroupRunsBy, (value: GroupRunsBy) => void] => {
-  const validate = useCallback(
-    (value: string) => {
-      switch (value) {
-        case 'job':
-        case 'automation':
-          return value;
-        default:
-          return defaultValue;
-      }
-    },
-    [defaultValue],
-  );
+  const [storedValue, setStoredValue] = useStateWithStorage(GROUP_BY_KEY, validate);
 
-  const [groupRunsBy, setGroupRunsBy] = useQueryAndLocalStoragePersistedState<GroupRunsBy>({
-    localStorageKey: GROUP_BY_KEY,
+  const [queryValue, setQueryValue] = useQueryPersistedState({
     queryKey: 'groupBy',
-    encode: (value) => {
-      return {groupBy: value};
-    },
-    decode: (pair) => {
-      return validate(pair.groupBy);
-    },
-    isEmptyState: (value) => !value,
```

Comment:
As this is currently written, `isEmptyState` is never true, because `decode` always returns a value via `validate`. So the initial (stored) value is never used.

It seems like this is probably a bug with the callsite rather than the hook itself, but as I noted in the summary, I'm not entirely sure that the current behavior of `useQueryAndLocalStoragePersistedState` is actually what we want here anyway. I don't think this callsite needs to write anything to the query state on initial render, and I don't think it needs anything with "empty states", but I don't know if that's true of other callsites. (I find the hook a little hard to follow, so apologies if I'm misunderstanding it.)